var/spack/repos/builtin/packages/nekbone/package.py

```diff
@@ -44,7 +51,10 @@ def install(self, spec, prefix):
 
         for wdir in working_dirs:
             with working_dir('test/' + wdir):
-                makenek = Executable('./makenek')
+                makenec = FileFilter('makenek')
+                makenec.filter('CC.*', 'CC='+self.spec['mpi'].mpicc)
+                makenec.filter('FF77.*', 'FF77='+self.spec['mpi'].mpif77)
+                makenek = Executable('./makenek~')
```

Comment:
This is the backup of the original file, without the patches. Is this what you want?